### PR TITLE
Allow passing in a promise that resolves to a request cache strategy for dictionary load

### LINKED ISSUE #417: Object Provider should allow dictionary load to pass cache strategy in request
**Is your feature request related to a problem? Please describe.**
When the dictionary updates, Open MCT should force a cache refresh and load the dictionary from the request.

**Describe the solution you'd like**
Allow more control over cache strategy.

**Testing Instructions**
see related VIPER ticket

**Additional context**
Has related VIPER ticket

### PR DESCRIPTION
**Pull request by davetsay**
Closes #417

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/providers/limit-provider.js
Comment (davetsay) [inline src/providers/limit-provider.js:1]: This was a fix for a lint error that snuck in from a different, unrelated, PR.

File: src/providers/object-provider.js [hunk 1]
```diff
@@ -180,7 +181,7 @@ export default class YamcsObjectProvider {
 
     #getTelemetryDictionary() {
         if (!this.dictionaryPromise) {
-            this.dictionaryPromise = this.#loadTelemetryDictionary(this.url, this.instance, this.folderName)
+            this.dictionaryPromise = this.#loadTelemetryDictionary()
```
Comment (jvigliotta) [inline src/providers/object-provider.js:184]: 👍🏻

File: src/providers/object-provider.js [hunk 2]
```diff
@@ -193,8 +194,10 @@ export default class YamcsObjectProvider {
         const operation = 'parameters?details=yes&limit=1000';
         const parameterUrl = this.url + 'api/mdb/' + this.instance + '/' + operation;
         const url = this.#getMdbUrl('space-systems');
-        const spaceSystems = await accumulateResults(url, {}, 'spaceSystems', []);
-        const parameters = await accumulateResults(parameterUrl, {}, 'parameters', []);
+        const requestOptions = await this.dictionaryRequestCacheStrategyPromise;
```
Comment (jvigliotta) [inline src/providers/object-provider.js:197]: Should we have some other function like 'buildRequestOptions' that does this? I mean, we can always add it down the road, but it seems weird just assigning directly from the cache strategy promise.

**Review by jvigliotta — APPROVED** [review]
One small comment regarding clarity that could be addressed later once/if we get more request options being used. Otherwise looks good to me.

File: src/providers/object-provider.js (hunk 2 above)
Comment (davetsay) [reply src/providers/object-provider.js:197]: i hear you. we could change it now or wait until there are other requests. this one is very specific to loading the dictionary so i think we may be okay.

**Pull request merged**